src/Tizen.NUI.Gadget/Tizen.NUI/NUIGadget.cs

```diff
@@ -0,0 +1,265 @@
+﻿/*
+ * Copyright (c) 2023 Samsung Electronics Co., Ltd All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.ComponentModel;
+using Tizen.Applications;
+using Tizen.NUI.BaseComponents;
+
+namespace Tizen.NUI
+{
+    /// <summary>
+    /// This class represents a NUIGadget controlled lifecycles.
+    /// </summary>
+    /// <since_tizen> 11 </since_tizen>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class NUIGadget
```

Comment:
Why don't you add 'abstract' to prevent direct creation of instances?

Reply:
I thought if the developer does not override the OnCreate() method, it's meaningless. So I didn't consider adding the 'abstract' keyword to the class.
Done. Thanks.
I think adding 'abstract' keyword is better than the previous implementation.